platform-sdk/swirlds-jasperdb/src/test/java/com/swirlds/merkledb/files/DataFileCollectionTest.java

```diff
@@ -421,7 +423,8 @@ void merge(final FilesTestType testType) throws Exception {
                 System.out.println("DataFileCollectionTest.merge");
                 List<Path> mergedFiles = null;
                 try {
-                    List<DataFileReader<long[]>> filesToMerge = fileCollection.getAllCompletedFiles(MAX_TEST_FILE_MB);
+                    List<DataFileReader<?>> filesToMerge =
+                            (List<DataFileReader<?>>) (Object) fileCollection.getAllCompletedFiles(MAX_TEST_FILE_MB);
```

Comment:
This is double-cast.  Wouldn't it be better to just cast once?

Reply:
Unfortunately, this clumsy workaround has to be there because the compiler doesn't allow direct cast of `List<DataFileReader<long[]>>` to `List<DataFileReader<?>>`

Reply:
Why do we need the erasure?  Couldn't we just use the `List<DataFileReader<long[]>>` type

Reply:
_computer says no_ 

<img width="502" alt="image" src="https://github.com/hashgraph/hedera-services/assets/1595361/9a00afe8-d10f-4f93-b2ec-043c99a62190">

I think it has something to do with the fact that the wildcard is on the second level of depth. If it was `List<long[]` and `List<?>`, it would work.

Reply:
Generic capture is a serious pain, unfortunately. It always seems to create these ugly constructs...